tests/test_exponential_barycenter.py

```diff
@@ -0,0 +1,90 @@
+"""Unit tests for Exponential Barycenter mean."""
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.special_euclidean import SpecialEuclidean
+from geomstats.geometry.special_orthogonal import SpecialOrthogonal
+from geomstats.learning.exponential_barycenter import ExponentialBarycenter
+from geomstats.learning.frechet_mean import FrechetMean
+
+
+class TestExponentialBarycenter(geomstats.tests.TestCase):
+
+    def setUp(self):
+        self.se = SpecialEuclidean(n=3, point_type='matrix')
```

Comment:
Does the ExponentialBarycenter work with point_type='vector'?

The code in special_orthogonal and special_euclidean had the 'vector' implementation too, it will be good not to lose it, as it is probably faster than going to 'matrix', using ExponentialBarycenter, and converting back to 'vector'.

Reply:
It called FrechetMean in `special_orthogonal`, which is valid for the bi-invariant metric. For the `special_euclidean` case it called FrechetMean on the rotation part and the linear mean on the translation part (which I am not sure is valid). `point_type='vector'` is such a limited case of groups that I thought it was not relevant... But I can put it back if you insist

Reply:
And it was probably slower because FrechetMean calls `metric.squared_dist` and `metric.norm` for the stopping criterion, which is much slower than `gs.linalg.norm`used here. This is actually almost the only difference with FrechetMean

Reply:
I am not sure that the exponential barycenter formula for SE(3) is valid, especially since it seems that you didn't use the function `exponential_matrix`, which is needed to link the rotation part with the translation part (see original implementation).

Reply:
I agree that `point_type='vector'` is a limited case for Lie groups when you take the point of view of the mathematician and consider all Lie groups which equal weights.

However, from an applied perspective, `point_type='vector'` applies to SO(3) and SE(3) which are probably the most widely used Lie groups. The vector representation might:
- speed-up the computations,
- decrease the memory space needed,
- or simply be the favorite representation for users computing with SO(3) and SE(3) - and we might have many users in this case -
which I think are 3 good reasons to keep them. What do you think?

Reply:
> I am not sure that the exponential barycenter formula for SE(3) is valid, especially since it seems that you didn't use the function `exponential_matrix`, which is needed to link the rotation part with the translation part (see original implementation).

In matrix representation this is automatically done by the matrix exponential of the squared matrix of size n+1

Reply:
Got it!

Reply:
> I agree that `point_type='vector'` is a limited case for Lie groups when you take the point of view of the mathematician and consider all Lie groups which equal weights.
> 
> However, from an applied perspective, `point_type='vector'` applies to SO(3) and SE(3) which are probably the most widely used Lie groups. The vector representation might:
> 
>     * speed-up the computations,
> 
>     * decrease the memory space needed,
> 
>     * or simply be the favorite representation for users computing with SO(3) and SE(3) - and we might have many users in this case -
>       which I think are 3 good reasons to keep them. What do you think?

Ok I agree, but we need to separate the vector and matrix implementations of these groups because they become confusing being so long and not dealing with the matrix type all the time (there were many e.g. `assert logs.ndim == 2` which crash when using matrices, etc).

Reply:
Definitely agreed to this! It also goes back to @opeltre remark that the special_orthogonal was full with computations for the n=3 case, see PR #484.